### DISCUSSION TITLE
multi-platform project: common repository, or separate, individual repositories for each platform

### DISCUSSION THREAD

**Discussion by Papa-26** [community/community#82699] [New to GitHub] [answered]
I have a large running, almost completed, multi-platform, open source project. Up to now I have used a common repository, whith commented-out options. Since code-signing and notarization for Apple involves more complex instructions, should I use separate repositories, reserved branches, or keep it on the same, single-branch repository?

**Comment by faridnec** [comment]
Considering your project is almost completed, and assuming you've got good documentation, having separate branches in a single repository might strike a nice balance. You keep everything in one place but maintain a bit of separation for platform-specific concerns.

if you separate, it may add some redundancy if there's a lot of shared code and maintenance might require updates in multiple repositories. it is a hell of a work, to be honest. If you do reserved branches, it might get crowded in the long run, especially if there are many platform-specific considerations. Though, the decision is up to you, hope this helps to be taken into consideration

**Reply by Papa-26** [reply]
Thank you for your reply. That raises another question: let's say that I want to build for a specific platform, how do I tell e.g. Intellij, which branches to use, respectively, which to ignore?

**Reply by faridnec** [reply, marked as answer]
most case main or master branch -whichever you've designated as the primary development branch- is typically what IntelliJ will track by default. Any changes you make will be based on this branch and IntelliJ integrates well with version control systems. You can use the "Version Control" tool window to see changes across branches, commit your changes, and switch branches easily. For more info u can check [this documentation](https://www.jetbrains.com/help/idea/work-on-several-features-simultaneously.html#feature-branches) especially under the ***Use feature branches*** section

**Reply by Papa-26** [reply]
Thank you for your help
